source/Grid/Grid.js

```diff
@@ -597,29 +597,26 @@ class Grid extends React.PureComponent<Props, State> {
     // In that event we need to remeasure.
     if (!instanceProps.scrollbarSizeMeasured) {
       this.setState(prevState => {
-        prevState.instanceProps.scrollbarSize = getScrollbarSize();
-        prevState.instanceProps.scrollbarSizeMeasured = true;
-        prevState.needToResetStyleCache = false;
-        return prevState;
+        const stateUpdate = {...prevState, needToResetStyleCache: false};
+        stateUpdate.instanceProps.scrollbarSize = getScrollbarSize();
+        stateUpdate.instanceProps.scrollbarSizeMeasured = true;
+        return stateUpdate;
```

Comment:
Is it ok here to mutate state?

Reply:
I think so. previously it was [this](https://github.com/bvaughn/react-virtualized/blob/2a7a82813f34bcce3f6d52036c5a96d538041687/source/Grid/Grid.js#L536:L539)

And since I need to set stuff on `instanceProps` I use prevState as a base